### PR TITLE
Cherry pick favorite team to master

### PR DESCRIPTION
**Pull request by winprn**
# Description
Type of change: new feature
Note: since the user table has been changed, from `id="users-table"` to `class="users-table`, I made some fixes to render everything properly

## What

What does the PR do?
It moves some commits from icpc22 into master

# Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have explained the purpose of this PR.
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the README/documentation
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Informed of breaking changes, testing and migrations (if applicable).
- [ ] Attached screenshots (if applicable).

By submitting this pull request, I confirm that my contribution is made under the terms of the AGPL-3.0 License.

### CODE REVIEW

File: templates/contest/ranking.html [hunk 1]
```diff
@@ -356,6 +475,7 @@
             };
 
             window.applyRankingFilter();
+            console.log('hello');
```
Comment (hieplpvip) [inline templates/contest/ranking.html:478]: Remove this

File: templates/contest/ranking.html [hunk 2]
```diff
@@ -364,6 +484,141 @@
...
+                    table_body.insertBefore(new_row.get(0), table_body.children[index++]);
+                });
+
+                $(table_body.children[index - 1]).css('border-bottom', 'thick solid black');
+            }
+
+            window.applyFavoriteUsers();
+
+            $(document).ready(() => {
+                console.log('done');
```
Comment (hieplpvip) [inline templates/contest/ranking.html:521]: remove this

**Comment by leduythuccs** [comment]
don't merge this yet

**Comment by leduythuccs** [comment]
will close this

**Pull request closed without merging**